2015-Fall/nativix/Calculator/Calculator/Calculator/ViewController.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  ViewController.swift
+//  Calculator
+//
+//  Created by MARIA NATIVIDAD LARA DIAZ on 01/09/15.
+//  Copyright (c) 2015 Appstract Development. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+
+    @IBOutlet weak var display: UILabel!
+    
+    var userIsInTheMiddleOfTypingANumber = false
+   
+    @IBAction func appendDigit(sender: UIButton) {
+        let digit = sender.currentTitle!
+        if userIsInTheMiddleOfTypingANumber {
+            display.text = display.text! + digit
+        } else {
+            display.text = digit
+            userIsInTheMiddleOfTypingANumber = true
+        }
+    }
+    
+    var numberStack = [Double]()
+    var symbolStack = [String]()
+    
+    @IBAction func operate(sender: UIButton) {
+        userIsInTheMiddleOfTypingANumber = false
+        numberStack.append((display.text! as NSString).doubleValue)
+        println("numberStack = \(numberStack)")
+        if symbolStack.count >= 1 && numberStack.count >= 1 {
+            let operation = symbolStack.removeLast()
+            switch operation {
+            case "+" : performOperation{ $0 + $1}
```

Comment:
You can replace this (and a few below) with a identical function that is already defined. In this case with `+`. Please ask next time if this is not clear why.